portfolio/src/test/java/com/google/sps/BookResponseParserTest.java

```
@@ -0,0 +1,94 @@
+        +"mighty armies grapple for power over the one thing that could destroy them all.";
+
+    Book.Builder builder = Book.builder().title(title)
+            .categories(categories).authors(authors).language(language)
+            .description(description).infoLink(infoLink).pageCount(pageCount)
+            .publishedDate(publishedDate).publisher(publisher)
+            .maturityRating(maturityRating).thumbnail(thumbnail).isbn(isbn);
+    Book expected = builder.build();
+
+    String fileString = "";
+    try {
+      File file = new File("src/test/java/com/google/sps/BookParserTestFile.csv");
+      Scanner scanner = new Scanner(file, "utf-8");
+      while(scanner.hasNextLine()){
+          fileString += scanner.nextLine();
+      }
+    } catch (Exception ex) {
+      Assert.fail("Could not open file. " + ex);
+    }
+
+    try{
+        if(!fileString.equals("")){
+            Book actual = BookResponseParser.parseBook(fileString);
+            Assert.assertEquals(expected.title(),actual.title());
+            Assert.assertEquals(expected.genre(),actual.genre());
+            Assert.assertEquals(expected.categories(),actual.categories());
+            Assert.assertEquals(expected.authors(),actual.authors());
+            Assert.assertEquals(expected.language(),actual.language());
+            Assert.assertEquals(expected.description(),actual.description());
+            Assert.assertEquals(expected.infoLink(),actual.infoLink());
+            Assert.assertEquals(expected.pageCount(),actual.pageCount());
+            Assert.assertEquals(expected.publishedDate(),actual.publishedDate());
+            Assert.assertEquals(expected.publisher(),actual.publisher());
+            Assert.assertEquals(expected.maturityRating(),actual.maturityRating());
+            Assert.assertEquals(expected.thumbnail(),actual.thumbnail());
+            Assert.assertEquals(expected.isbn(),actual.isbn());
+        }
+    }
+    catch(Exception e){
+        e.printStackTrace();
```

in case there is an exception, this would hide the exception, and the test case would fail. So, better way would be to not catch the exception.